spec/models/user_spec.rb

```diff
@@ -1997,12 +1997,35 @@ def filter_by(method)
       expect(user.user_profile.reload.badge_granted_title).to eq(false)
 
       badge.update!(allow_title: true)
+      user.badges.reload
```

Comment:
We don't need to reload the badges here. Since we only set the title of user.

Reply:
I do need to do that because the spec fails without doing so. The `User#check_if_title_is_badged_granted` method uses ruby `Enumerable#find` to save trip to the DB https://devdocs.io/ruby~2.6/enumerable#method-i-find.

```
badge_matching_title = title.present? && badges.find do |badge|
  badge.allow_title? && (badge.display_name == title || badge.name == title)
end
```

Because the user's `badges` AR association is already loaded it will not query the DB again to fetch the updated `allow_title`. I can alternatively call `user.badges.find { |b| b.id == badge.id }.update!(allow_title: true)` but I think that's much of a muchness really. Or I can change the User model code to fetch the badges from the db instead of using `badges.find`. Is there a downside to reloading that I'm not aware of?

Reply:
@vinothkannans